packages/editor-ui/src/stores/apiKeys.store.ts

```diff
@@ -0,0 +1,65 @@
+import { STORES } from '@/constants';
+import type { ApiKey } from '@/Interface';
+import { defineStore } from 'pinia';
+import { useRootStore } from '@/stores/root.store';
+
+import * as publicApiApi from '@/api/api-keys';
+import { computed, ref } from 'vue';
+import { useSettingsStore } from './settings.store';
+
+export const useApiKeysStore = defineStore(STORES.API_KEYS, () => {
+	const apiKeys = ref<ApiKey[]>([]);
+
+	const rootStore = useRootStore();
+	const settingsStore = useSettingsStore();
+
+	const apiKeysSortByCreationDate = computed(() =>
+		apiKeys.value.sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
+	);
+
+	const apiKeysById = computed(() => {
+		return apiKeys.value.reduce(
+			(acc, apiKey) => {
+				acc[apiKey.id] = apiKey;
+				return acc;
+			},
+			{} as Record<string, ApiKey>,
+		);
+	});
+
+	const canAddMoreApiKeys = computed(() => apiKeys.value.length < settingsStore.api.apiKeysLimit);
+
+	const getAllApiKeys = async () => {
+		if (apiKeys.value.length) return apiKeys.value;
+		apiKeys.value = await publicApiApi.getApiKeys(rootStore.restApiContext);
+		return apiKeys.value;
```

Comment:
This can be a bit confusing as it does a request only if we don't have the results cached. Can we add a JSDoc comment about or rename it to reflect this?